build.rs

```diff
@@ -1,34 +1,156 @@
+use std::env;
+use std::path::Path;
+use std::path::PathBuf;
+use std::process::Command;
+
+#[cfg(feature = "bundled")]
+fn fetch_hwloc(path: &Path, version: &str) {
+    if !path.join("hwloc").join("Makefile.am").exists() {
+        let r = Command::new("git")
+            .args(&[
+                "clone",
+                "https://github.com/open-mpi/hwloc",
+                "--depth",
+                "1",
+                "--branch",
+                version,
+            ])
+            .current_dir(path.clone())
+            .output();
+        match r {
+            Ok(m) => {
+                if !m.status.success() {
+                    eprintln!("git clone for hwloc returned {}", m.status);
+                    eprintln!("{:?}", m);
+                }
+            }
+            Err(e) => eprintln!("Error cloning hwloc {}", e),
+        }
+    }
+}
+
+#[cfg(feature = "bundled")]
+fn get_os_from_triple(triple: &str) -> Option<&str> {
+    triple.splitn(3, "-").nth(2)
+}
+
+#[cfg(feature = "bundled")]
+fn compile_hwloc2_autotools(p: PathBuf) -> PathBuf {
+    let mut config = autotools::Config::new(p);
+    config.reconf("-ivf").make_target("install").build()
+}
+
+fn compile_hwloc2_cmake(build_path: &Path, target_os: &str) -> PathBuf {
+    let mut cfg = cmake::Config::new(build_path);
+    if let Ok(profile) = env::var("HWLOC_BUILD_PROFILE") {
+        cfg.profile(&profile);
+    } else {
+        cfg.profile("release");
+    }
+
+    // Allow specifying custom toolchain specifically for SDL2.
+    if let Ok(toolchain) = env::var("HWLOC_TOOLCHAIN") {
+        cfg.define("CMAKE_TOOLCHAIN_FILE", &toolchain);
+    } else {
+        // Override __FLTUSED__ to keep the _fltused symbol from getting defined in the static build.
+        // This conflicts and fails to link properly when building statically on Windows, likely due to
+        // COMDAT conflicts/breakage happening somewhere.
+        #[cfg(feature = "static-link")]
+        cfg.cflag("-D__FLTUSED__");
+
+        #[cfg(target_os = "linux")]
+        {
+            // Add common flag for affected version and above
+            use version_compare::{compare_to, Cmp};
+            if let Ok(version) = std::process::Command::new("cc")
+                .arg("-dumpversion")
+                .output()
+            {
+                let affected =
+                    compare_to(std::str::from_utf8(&version.stdout).unwrap(), "10", Cmp::Ge)
+                        .unwrap_or(true);
+                if affected {
+                    cfg.cflag("-fcommon");
+                }
+            }
+        }
```

Comment:
This does not make sense:

- It is only build on Linux, while CMake is only used on Windows
- It assumes that the compiler has a `-dumpversion` switch, which may not be true.
- It compares that version to 10 without knowing what the compiler is (GCC v10 and clang v10 definitely are not the same thing).

Since CMake is not used on Linux, I would just delete this code. And then you can drop the version_compare dependency along the way.